app/assets/javascripts/project_find_file.js.coffee

```diff
@@ -0,0 +1,109 @@
+class @ProjectFindFile
+  constructor: (@options)->
+    @filePaths = {}
+    @element = ".file-finder-input"
+    @row = "<tr class='tree-item'><td class='tree-item-file-name'><i class='fa fa-file-text-o fa-fw'></i><span class='str-truncated'><a href=%blobItemUrl>%filePath</a></span></td></tr>"

```

Comment:
Please build this structure using jQuery. Doing it as a string opens us up to XSS.